oidc-controller/api/templates/verified_credentials.html

```diff
@@ -333,193 +287,237 @@ <h1 class="mobile-device my-1">Continue with:</h1>
-          setUiElements(".success", false, true, false);
-          setTimeout(() => {
-            window.location.replace("{{callback_url}}", { method: "POST" });
-          }, 2000);
-          break;
-        case "failed":
-          setUiElements(".failed", true, false, false);
-          break;
-        case "pending":
-          setUiElements(".pending", false, true, true);
-          break;
-        case "expired":
-          setUiElements(".expired", true, false, false);
-          break;
-        case "abandoned":
-          setUiElements(".abandoned", true, false, false);
-          break;
-      }
-    };
+      /**
+       * @typedef  {"intro" |"verified" | "failed" | "pending"| "expired" |"abandoned"} AppStates
+       */
+     const state = reactive({
+       /**
+        * @type {AppStates}
+        */
+       current: "intro",
+       showSpinner: false,
+       showScanned: false,
+       showRefresh: false,
+       disableDeeplink: false,
+     });
+     const getBrowser = () => {
+       let userAgent = navigator.userAgent || navigator.vendor;
 
-    document.getElementById("refresh-button").addEventListener("click", () => {
-      location.reload(true);
-    });
+       if (/android/i.test(userAgent)) {
+         return "Android";
+       }
 
-    /**
-     * Set the UI elements based on the current state
-     */
-    const setUiElements = (state, showRefresh, showScanned, setSpinner) => {
-      console.log("Setting UI elements");
-      const stateElement = document.querySelectorAll(state);
-      const qrcode = document.querySelector(".qr-code");
-      const scannedMask = document.querySelector(".scanned-mask");
-      const refreshButton = document.getElementById("refresh-button");
+       if (
+         /iPad|iPhone|iPod/.test(userAgent) ||
+         (/Macintosh/.test(userAgent) && "ontouchend" in document) ||
+         (navigator.platform === "MacIntel" && navigator.maxTouchPoints > 1) ||
+         (navigator.vendor && navigator.vendor.indexOf("Apple") > -1)
+       ) {
+         return "iOS";
+       }
 
-      // set all elements to display: none and display the current state
-      document.querySelectorAll(".header-desc").forEach((el) => {
-        el.style.display = "none";
-      });
-      stateElement.forEach((el) => {
-        el.style.display = "grid";
-      });
+       return "unknown";
+     };
 
-      // For expired or success state, set the deep link to disabed
-      if (state === ".expired" || state === ".success") {
-        const deepLink = document.getElementById("deep-link-button");
-        deepLink.classList.add("disabled");
-        deepLink.href = "";
-      }
+     createApp({
+       data() {
+         const mobileDevice = ref(
+           getBrowser() === "Android" || getBrowser() === "iOS" ? true : false
+         );
+      const errorMessage = computed(() => {
+      if (state.current === "failed") {
+      return `<b>We couldn't verify your credentials because they may be:</b>
```

Comment:
Can these not all be Vue components themselves, now?

Reply:
Couldn't agree more. Once I'm back I will extract it. For whatever reason I couldn't find a way to do it without multiple files or a build step but I was able to dig up a solution.